owncloudApp/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java

```diff
@@ -1309,38 +1310,25 @@ private Cursor getShareCursorForValue(String key, String value) {
-            share.setFileSource(c.getLong(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_ITEM_SOURCE)));
-            share.setShareType(ShareType.fromValue(c.getInt(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARE_TYPE))));
-            share.setShareWith(c.getString(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARE_WITH)));
-            share.setPermissions(c.getInt(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_PERMISSIONS)));
-            share.setSharedDate(c.getLong(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARED_DATE)));
-            share.setExpirationDate(c.getLong(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_EXPIRATION_DATE)));
-            share.setToken(c.getString(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_TOKEN)));
-            share.setSharedWithDisplayName(c.getString(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARE_WITH_DISPLAY_NAME)));
-            share.setSharedWithAdditionalInfo(c.getString(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARE_WITH_ADDITIONAL_INFO)));
-            share.setIsFolder(c.getInt(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_IS_DIRECTORY)) == 1);
-            share.setUserId(c.getLong(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_USER_ID)));
-            share.setIdRemoteShared(c.getLong(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_ID_REMOTE_SHARED)));
-            share.setName(c.getString(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_NAME)));
-            share.setShareLink(c.getString(
-                    c.getColumnIndex(ProviderTableMeta.OCSHARES_URL)));
+            share = new OCShare(
+                    c.getLong(c.getColumnIndex(ProviderTableMeta.OCSHARES_FILE_SOURCE)),
+                    c.getLong(c.getColumnIndex(ProviderTableMeta.OCSHARES_ITEM_SOURCE)),
+                    c.getInt(c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARE_TYPE)),
+                    c.getString(c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARE_WITH)),
+                    c.getString(c.getColumnIndex(ProviderTableMeta.OCSHARES_PATH)),
+                    c.getInt(c.getColumnIndex(ProviderTableMeta.OCSHARES_PERMISSIONS)),
+                    c.getLong(c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARED_DATE)),
+                    c.getLong(c.getColumnIndex(ProviderTableMeta.OCSHARES_EXPIRATION_DATE)),
+                    c.getString(c.getColumnIndex(ProviderTableMeta.OCSHARES_TOKEN)),
+                    c.getString(c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARE_WITH_DISPLAY_NAME)),
+                    c.getString(c.getColumnIndex(ProviderTableMeta.OCSHARES_SHARE_WITH_ADDITIONAL_INFO)),
+                    c.getInt(c.getColumnIndex(ProviderTableMeta.OCSHARES_IS_DIRECTORY)) == 1,
```

Comment:
In library project, I see you have used a constant to determine if it is a folder or not. Can that constant be used here?

Reply:
Where have you seen that? [RemoteShare.kt](https://github.com/owncloud/android-library/blob/new_arch/get_public_shares_for_file/owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt#L54) has just this

    var isFolder: Boolean = false

Reply:
I don't remember... So many changes....